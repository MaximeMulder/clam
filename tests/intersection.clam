type AB = {a: Int, c: String} & {a: Int, b: String};

def main = {
    var ab: AB = {a = 1, b = "Hello", c = "World"};
    var a: Int = ab.a;
    var b: String = ab.b;
    var c: String = ab.c;
    print(ab);
    print(a);
    print(b);
    print(c);
    unit
};
