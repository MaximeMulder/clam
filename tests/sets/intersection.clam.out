{a = 1, b = "Hello", c = "World"}
1
"Hello"
"World"
